packages/commonwealth/server/routes/discord/setDiscordBotConfig.ts

```diff
@@ -91,29 +91,38 @@ const setDiscordBotConfig = async (
     where: { id: community_id },
   });
 
+  const transaction = await models.sequelize.transaction();
   if (
     existingCommunityWithGuildConnected &&
     existingCommunityWithGuildConnected.length > 0
   ) {
     // Handle discord already linked to another CW community
     communityInstance.discord_config_id = null;
-    await communityInstance.save();
+    await communityInstance.save({ transaction });
 
     await models.DiscordBotConfig.destroy({
       where: {
         community_id,
       },
+      transaction,
     });
     log.info(
       'Attempted to add a guild that was already connected to another CW community.',
     );
 
+    try {
+      await transaction.commit();
+    } catch (err) {
+      await transaction.rollback();
+    }
```

Comment:
Is there a reason not to wrap the previous logic from L99 in the try block?

Reply:
Good catch - this would leave txn hanging if something failed.